src/middlewared/middlewared/plugins/device_/vrrp_events_linux.py

```diff
@@ -0,0 +1,57 @@
+import os
+import time
+
+from middlewared.utils import start_daemon_thread
+
+VRRP_FIFO_FILE = '/var/run/vrrpd.fifo'
+TIMEOUT = 2
+VRRP_FIFO_CONNECTED = False
+
+
+def vrrp_hook_license_update(middleware, prev_product_type, *args, **kwargs):
+
+    # get new product_type
+    new_product_type = middleware.call_sync('system.product_type')
+
+    if prev_product_type != 'SCALE_ENTERPRISE':
+        if new_product_type == 'SCALE_ENTERPRISE':
+            vrrp_fifo_listen(middleware)
```

Comment:
Hmm, not exactly what I had in mind as we still do not keep track of a running thread and its not clear how this is handled within the event loop as its a blocking call to a thread...


how about:

VRRP_THREAD = None

...

global VRRP_THREAD
if licensed:
    VRRP_THREAD = start_daemon_thread(...)


...


global VRRP_THREAD
if prod_type != new_prod_type and VRRP_THREAD is None:
    VRRP_THREAD = start_daemon_thread(...)